File: SeleniumExercise/src/main/java/api/resources/FoldersResource.java
package api.resources;

import api.config.APIAuth;
import api.dtos.FolderDTO;
import io.restassured.specification.RequestSpecification;
import static api.AuthorizationHeaders.venaBasic;
import static io.restassured.RestAssured.given;
import static org.apache.http.HttpHeaders.AUTHORIZATION;

public class FoldersResource {
	public static FolderDTO[] list(APIAuth auth) {
		return list(auth, null);
	}

	public static FolderDTO[] list(APIAuth auth, Boolean includeHidden) {
		RequestSpecification spec = given()
				.header(AUTHORIZATION, venaBasic(auth));
		if (includeHidden != null) {
			spec = spec.queryParam("includeHidden", includeHidden);
		}
		return spec
				.get("/api/manager/folders")
				.then()
				.statusCode(200)
				.extract()
				.as(FolderDTO[].class);
	}

	public static void delete(APIAuth auth, long folderId) {
		given()
				.header(AUTHORIZATION, venaBasic(auth))
				.pathParam("folderId", folderId)
				.delete("/api/manager/folders/{folderId}")
				.then()
				.statusCode(204);
	}
}

File: SeleniumExerciseC#/api/resources/FoldersResource.cs
using API;
using config;
using RestSharp;
using System.Text.Json;

namespace resources
{
    public class FoldersResource
    {
        public static FolderDTO[] list(IAuth auth, Boolean includeHidden = false)
        {
            RestClient client = new RestClient("https://dev.vena.io/api/manager/folders");
            RestRequest request = new RestRequest();
            request.AddHeader("Authorization", AuthorizationHeaders.venaBasic(auth));
            Task<RestResponse> task = client.GetAsync(request);
            task.Wait(TimeSpan.FromSeconds(5));
            return JsonSerializer.Deserialize<FolderDTO[]>(task.Result.Content);
        }

        public static void delete(IAuth auth, string folderId)
        {
            RestClient client = new RestClient("https://dev.vena.io/api/manager/folders/{folderId}");
            RestRequest request = new RestRequest();
            request.AddHeader("Authorization", AuthorizationHeaders.venaBasic(auth));
            request.AddParameter("folderId", folderId, ParameterType.UrlSegment);
            Task<RestResponse> task = client.DeleteAsync(request);
            task.Wait(TimeSpan.FromSeconds(5));
        }
    }
}